### PR TITLE
Support updating the NMS decoder in YOLOv8

### LINKED ISSUE #1840: YOLOv8 does not support updating the `prediction_decoder`
If you build a YOLOV8Detector model `yolov8` and call

`yolov8.prediction_decoder = some_new_prediction_decoder`, the predict function isn't being updated correctly.

### PR DESCRIPTION
**Pull request by ianstenbit**
Fixes #1840.

cc @innat

### CODE REVIEW

File: keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py [hunk 1]
```diff
@@ -608,14 +608,34 @@ def decode_predictions(
...
+                "use the same `bounding_box_format`, but got "
+                "`prediction_decoder.bounding_box_format="
+                f"{prediction_decoder.bounding_box_format}`, and "
+                "`self.bounding_box_format="
+                f"{self.bounding_box_format}`."
+            )
+        self._prediction_decoder = prediction_decoder
+        self.make_predict_function(force=True)
+        self.make_train_function(force=True)
+        self.make_test_function(force=True)
```
Comment (haifeng-jin) [inline keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py:629]: Don't we need the return values of these functions?

File: keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py [hunk 2]
```diff
@@ -608,14 +608,34 @@ def decode_predictions(
...
+
     def get_config(self):
         return {
             "num_classes": self.num_classes,
             "bounding_box_format": self.bounding_box_format,
             "fpn_depth": self.fpn_depth,
             "backbone": keras.utils.serialize_keras_object(self.backbone),
             "label_encoder": self.label_encoder,
-            "prediction_decoder": self.prediction_decoder,
+            "prediction_decoder": self._prediction_decoder,
```
Comment (haifeng-jin) [inline keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py:638]: Do we need to update this? Seems directly using the property is safer.

**Review by haifeng-jin — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py:638]: I've just copied what we did in RetinaNet -- I think it doesn't particularly matter but if there is a reason I should prefer the property I'm happy to switch it in both models. wdyt?

File: keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py:629]: No, these just need to get called to update the graph. No need to store the results here, as they'll be stored on the model in the call to `make_xyz_function`

**Review by haifeng-jin — APPROVED** [review]

**Pull request merged**